core/utils_test.py

```diff
@@ -794,3 +794,85 @@ def test_url_open(self) -> None:
         self.assertEqual(response.getcode(), 200) # type: ignore[attr-defined]
         self.assertEqual(
             response.url, 'http://www.google.com') # type: ignore[attr-defined]
+
+    def test_get_random_int(self) -> None:
+        self.assertLess(utils.get_random_int(5), 5)
+        self.assertGreaterEqual(utils.get_random_int(5), 0)
+        self.assertLess(utils.get_random_int(True), 1)
+        self.assertGreaterEqual(utils.get_random_int(True), 0)
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only positive integers allowed'):
+            utils.get_random_int(-1)
+
+    def test_get_random_choice(self) -> None:
+        list_instance = [1, 5, 9, 11, 15]
+        list_instance2: List[str] = []
+        self.assertIn(utils.get_random_choice(list_instance), list_instance)
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only non-empty lists allowed'):
+            utils.get_random_choice(list_instance2)
+
+    def test_get_human_readable_time_string(self) -> None:
+        """Should we test for negative time."""
+        self.assertEqual('December 12 06:42:12', utils.get_human_readable_time_string(944980932342.38))  # pylint: disable=line-too-long
+
+    def test_generate_new_session_id(self) -> None:
+        self.assertEqual(24, len(utils.generate_new_session_id()))
+        self.assertEqual(type(utils.generate_new_session_id()).__name__, 'str')
+
+    def test_require_valid_name_with_incorrect_input(self) -> None:
+        """ADD TEST ESCAPE SEQUENCE CHARACTERS CHECK."""
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            utils.ValidationError, 'The length of the exploration title should be between 1 and 50 ' 'characters; received '''):   # pylint: disable=line-too-long
```

Comment:
The triple quote at the end seems wrong

Reply:
It looks correct to me since the function looks like this:
![validName](https://user-images.githubusercontent.com/81075125/159530836-bb9ef119-dc41-4e85-8b29-f6a525bbddf7.JPG)
So when we give the input, name_type will be 'the exploration title' and name will be ' '
This will replace the format specifiers in the string given to give:
'The length of the exploration title should be between 1 and 50 ' 'characters; received '''

Reply:
In Python, putting strings next to each other inside parentheses concatenates them, so in your screenshot above, the full format string will be `'The length of %s should be between 1 and 50 characters; received %s'`. Then when we substitute in the arguments (`'the exploration title' and ''`) we'll get `'The length of the exploration title should be between 1 and 50 characters; received '`. Notice that when substituting in a string (empty or not), Python doesn't add quotation marks:

```python
>>> '[%s]' % 'test'
'[test]'
>>> '[%s]' % ''
'[]'
```

Given these issues, I'm surprised that this test passes. Can you try changing the test so that it should fail to make sure it's failing correctly? For example, change the string argument to `assertRaisesRegex` to be `test` since that should cause the test to fail

Reply:
I searched up online and from what I've understood it seems like ' ' and {{blank space}} are interchangeable so when we do 
'   ''' it is evaluating the '' nested inside to a blank. To summarize, ' ' and {{blank space}} are the same.

Reply:
It is not possible to do what you've suggested because changing '' to 'test' would apply for a different test case since len(name) < 1 won't be satisfied

Reply:
> I searched up online and from what I've understood it seems like ' ' and {{blank space}} are interchangeable so when we do ' ''' it is evaluating the '' nested inside to a blank. To summarize, ' ' and {{blank space}} are the same.

Ahh okay, I see why this works. It seems unnecessarily complicated though. Couldn't you just use `'The length of the exploration title should be between 1 and 50 characters; received '` as the expected error message? I tried this in a Python shell to confirm that this message is equivalent to what you have:

```python
>>> ('The length of the exploration title should be between 1 and 50 characters; received ') == ('The length of the exploration title should be between 1 and 50 ' 'characters; received ''')
True
```

Reply:
Yes, you're right it seems overly complicated. Should I change it to ```'The length of the exploration title should be between 1 and 50 characters; received '``` then?

Reply:
Yes, I think that would be clearer

Reply:
Done